src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs

```diff
@@ -108,7 +121,48 @@ public void Apply(Workspace workspace, Document fromDocument, IEnumerable<CodeAc
             var oldSolution = workspace.CurrentSolution;
             Solution updatedSolution = oldSolution;
 
-            foreach (var operation in operations)
+            var operationsList = operations.ToList();
+            if (operationsList.Count > 1)
+            {
+                // Make a linked undo to wrap all these operations.  This way we should
+                // be able to undo them all with one user action.
+                using (var transaction = workspace.OpenGlobalUndoTransaction(title))
```

Comment:
In general we had a policy in `VisualStudioWorkspace.ApplyChanges` of not creating a global undo unless we are changing more than one document, which you seem to be losing here.

We did this because committing a global undo saves all of the documents involved, which isn't what users expect for "simple" fixes that only change one document.